R/tm_outliers.R

```diff
@@ -241,7 +241,41 @@ srv_outliers <- function(id, data, reporter, filter_panel_api, outlier_var,
   checkmate::assert_class(data, "tdata")
   moduleServer(id, function(input, output, session) {
     vars <- list(outlier_var = outlier_var, categorical_var = categorical_var)
-    selector_list <- teal.transform::data_extract_multiple_srv(vars, data)
+
+    selector_list <- teal.transform::data_extract_multiple_srv(
+      data_extract = vars,
+      datasets = data,
+      select_validation_rule = list(
+        outlier_var = shinyvalidate::compose_rules(
+          shinyvalidate::sv_required("Please select a variable"),
+          ~ if ("categorical_var" %in% names(selector_list())) {
+            if (length(selector_list()$categorical_var()$select) > 0 &&
+              selector_list()$categorical_var()$select == (.)) {
+              "`Variable` and `Categorical factor` cannot be the same"
+            }
+          }
+        )
+      ),
+      filter_validation_rule = list(
+        categorical_var = shinyvalidate::compose_rules(
+          ~ if (length(selector_list()$categorical_var()$select) > 0 && length(.) == 0) {
```

Comment:
This checks length of `categorical_var` twice?

Reply:
not really, the first is for the selection of the variable and the second is for the levels.

Reply:
Isn't `selector_list()$categorical_var()$select` the value of the `categorical_var` input, just like `.`?

Reply:
alas no...
![image](https://user-images.githubusercontent.com/15201933/210343114-1d8afa85-dee5-4e4c-9c44-e7dc2160b3a7.png)

Reply:
Right. I changed the rule here but I cannot formulate the proper assertions in `categorical_var`.